tests/sentry/processing/realtime_metrics/test_redis.py

```diff
@@ -380,90 +392,104 @@ def test_projects_mixed_buckets(
+
+    assert list(counts) == [BucketedCount(timestamp=110, count=0)]
 
 
 def test_get_counts_for_project_empty(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time
 ) -> None:
     redis_cluster.set(
         "symbolicate_event_low_priority:counter:10:42:111",
         0,
     )
     redis_cluster.delete("symbolicate_event_low_priority:counter:10:42:111")
 
+    mock_time.return_value = 113.0
     counts = store.get_counts_for_project(42)
-    assert list(counts) == []
+
+    assert list(counts) == [BucketedCount(timestamp=110, count=0)]
 
 
 def test_get_counts_for_project_no_matching_keys(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time
 ) -> None:
     redis_cluster.set("symbolicate_event_low_priority:counter:10:53:111", 0)
 
+    mock_time.return_value = 113.0
     counts = store.get_counts_for_project(42)
-    assert list(counts) == []
+
+    assert list(counts) == [BucketedCount(timestamp=110, count=0)]
 
 
 def test_get_counts_for_project_negative_key(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time
 ) -> None:
-    redis_cluster.set("symbolicate_event_low_priority:counter:10:42:-111", 0)
+    redis_cluster.set("symbolicate_event_low_priority:counter:10:42:-110", 0)
 
+    mock_time.return_value = -103.0
     counts = store.get_counts_for_project(42)
+
     assert list(counts) == [
-        BucketedCount(timestamp=-111, count=0),
+        BucketedCount(timestamp=-110, count=0),
     ]
 
 
 def test_get_counts_for_project_negative_count(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time
 ) -> None:
-    redis_cluster.set("symbolicate_event_low_priority:counter:10:42:111", -10)
+    redis_cluster.set("symbolicate_event_low_priority:counter:10:42:110", -10)
 
+    mock_time.return_value = 113.0
     counts = store.get_counts_for_project(42)
+
     assert list(counts) == [
-        BucketedCount(timestamp=111, count=-10),
+        BucketedCount(timestamp=110, count=-10),
     ]
 
 
 def test_get_counts_for_project_multiple_projects(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time
 ) -> None:
-    redis_cluster.set("symbolicate_event_low_priority:counter:10:42:111", 0)
-    redis_cluster.set("symbolicate_event_low_priority:counter:10:42:222", 0)
-    redis_cluster.set("symbolicate_event_low_priority:counter:10:53:111", 0)
+    redis_cluster.set("symbolicate_event_low_priority:counter:10:42:110", 0)
+    redis_cluster.set("symbolicate_event_low_priority:counter:10:53:110", 0)
```

Comment:
since you've updated this test to focus primarily on multiple projects, could i bug you to add a test that checks that the hole-filling works? e.g. initialize two counters at 110 and 150 with nonzero counts, and then make sure that every bucket in between gets a `BucketedCount` entry?

Reply:
i'm guessing that in order to achieve this the store would need to have a different counter TTL... could i also make another request to extend the TTL for the `_empty` and/or the `_unset` tests to also verify that the time range calculations are correctly returning multiple bucketed counts when TTL > bucket size?

Reply:
Lol, a test for the exact thing I wanted to fix would probably be a good idea, yeah.